src/mod_smart_markers.erl

```diff
@@ -145,33 +152,56 @@ user_send_packet(Acc, _From, _To, _Packet) ->
 %%--------------------------------------------------------------------
 %% Other API
 %%--------------------------------------------------------------------
--spec get_chat_markers(ChatType :: chat_type(), To :: jid:jid(),
-                       Thread :: maybe_thread(), TS :: os:timestamp()) -> [chat_marker()].
-get_chat_markers(ChatType, #jid{lserver = LServer} = To, Thread, TS) ->
+-spec get_chat_markers(jid:jid(), maybe_thread(), integer()) ->
+    [chat_marker()].
+get_chat_markers(#jid{lserver = LServer} = To, Thread, TS) ->
     %% internal API, no room access rights verification here!
-    Host = case ChatType of
-               one2one -> LServer;
-               groupchat ->
-                   mod_muc_light_utils:muc_host_to_host_type(LServer)
-           end,
-    mod_smart_markers_backend:get_chat_markers(Host, To, Thread, TS).
+    {ok, HostType} = mongoose_domain_api:get_host_type(LServer),
+    mod_smart_markers_backend:get_chat_markers(HostType, To, Thread, TS).
 
 %%--------------------------------------------------------------------
 %% Local functions
 %%--------------------------------------------------------------------
--spec update_chat_markers(Acc :: mongoose_acc:t(),
-                          Host :: jid:lserver(),
-                          Markers :: [chat_marker()]) -> mongoose_acc:t().
-update_chat_markers(Acc, Host, Markers) ->
+-spec update_chat_markers(mongoose_acc:t(), mongooseim:host_type(), [chat_marker()]) ->
+    mongoose_acc:t().
+update_chat_markers(Acc, HostType, Markers) ->
     TS = mongoose_acc:timestamp(Acc),
-    [mod_smart_markers_backend:update_chat_marker(Host, CM) || CM <- Markers],
+    [mod_smart_markers_backend:update_chat_marker(HostType, CM) || CM <- Markers],
     mongoose_acc:set_permanent(?MODULE, timestamp, TS, Acc).
 
--spec extract_chat_markers(From :: jid:jid(),
-                           To :: jid:jid(),
-                           Packet :: exml:element(),
-                           TS :: integer()) -> [chat_marker()].
-extract_chat_markers(From, To, Packet, TS) ->
+-spec has_valid_markers(mongoose_acc:t(), jid:jid(), jid:jid(), exml:element()) ->
+    false | {true, mongooseim:host_type(), Markers :: [chat_marker()]}.
+has_valid_markers(Acc, From, To, Packet) ->
+    case is_valid_markers(Acc, From, To, Packet) of
+        false -> false;
+        {true, Markers} ->
```

Comment:
Minor: I would get rid of `is_valid_markers`. Apart form the grammatical error (that was there before), it does not bring any extra value.
```suggestion
    case extract_chat_markers(Acc, From, To, Packet) of
        [] -> false;
        Markers ->
```